### PR TITLE
Use local details cleaner

### PR DESCRIPTION
**Pull request by isaacseymour**
Builds on #13, and includes that, so don't review until that's merged!

### CODE REVIEW

File: lib/ibandit/iban.rb [hunk 1]
```diff
@@ -186,13 +186,14 @@ def decomposable?
     end
 
     def build_iban_from_local_details(details_hash)
-      local_details = details_hash.dup
+      local_details = LocalDetailsCleaner.clean(details_hash)
```
Comment (greysteil) [inline lib/ibandit/iban.rb]: Slightly worried about mutating here

Comment (isaacseymour) [inline lib/ibandit/iban.rb]: The cleaner always does a `details_hash.merge(cleaned_stuff)`, so will return a different hash. The individual strings within in could still get mutated though, which would be annoying. I'll `.dup` them all.

File: lib/ibandit/iban_assembler.rb [hunk 2]
```diff
@@ -39,10 +46,7 @@ def self.assemble_at_bban(opts)
       #
       # Padding:
       #   Add leading zeros to account number if < 11 digits.
-      [
-        opts[:bank_code],
-        opts[:account_number].rjust(11, '0')
-      ].join
+      [opts[:bank_code], opts[:account_number]].join
```
Comment (greysteil) [inline lib/ibandit/iban_assembler.rb]: Almost all of these methods are just a join. Should we do that by default and only hadle exceptions (Italy) separately?

File: lib/ibandit/iban_assembler.rb [hunk 3]
```diff
@@ -39,10 +46,7 @@ def self.assemble_at_bban(opts)
       #
       # Padding:
       #   Add leading zeros to account number if < 11 digits.
```
Comment (greysteil) [inline lib/ibandit/iban_assembler.rb]: These comments (especially padding) are more relevant to the `LocalDetailsCleaner`. Move them in a separate PR?

**Comment by greysteil** [comment]
:+1:

**Comment by greysteil** [comment]
@isaacseymour - I've updated based on my comments. Can you take a look?

File: lib/ibandit/iban.rb [hunk 4]
```diff
@@ -186,13 +186,14 @@ def decomposable?
     end
 
     def build_iban_from_local_details(details_hash)
-      local_details = details_hash.dup
+      local_details = LocalDetailsCleaner.clean(details_hash.dup)
```
Comment (isaacseymour) [inline lib/ibandit/iban.rb]: This won't be enough :(
```ruby
hash = { a: 'thing' }
hash2 = hash.dup
hash2[:a].gsub!('thing', 'other thing')
hash # => { :a => 'other thing' }
```

Comment (greysteil) [inline lib/ibandit/iban.rb]: :( Are you OK to take back control of the PR?

**Comment by isaacseymour** [comment]
@greysteil I think this one is ready to merge now, if you're :+1: on the last few commits

File: lib/ibandit/iban.rb [hunk 5]
```diff
@@ -205,6 +204,12 @@ def extract_local_details_from_iban!
       @account_number = local_details[:account_number]
     end
 
+    def try_dup(thing)
+      thing.dup
```
Comment (greysteil) [inline lib/ibandit/iban.rb]: `obj` ?

File: lib/ibandit/iban_assembler.rb [hunk 6]
```diff
@@ -502,37 +65,53 @@ def self.assemble_sm_bban(opts)
...
+    def self.can_assemble?(local_details)
+      SUPPORTED_COUNTRY_CODES.include?(local_details[:country_code]) &&
+        required_fields?(local_details)
+    end
 
-        raise ArgumentError,
-              "#{arg} is a required field when assembling an " \
-              "#{country_code} IBAN"
-      end
+    def self.required_fields?(local_details)
```
Comment (greysteil) [inline lib/ibandit/iban_assembler.rb]: rename to `valid_arguments?` or something like that.

**Comment by greysteil** [comment]
:+1:
